transaction/src/manifest/parser.rs

```diff
@@ -812,14 +913,20 @@ impl Parser {
                 }
             }
             _ => {
-                return Err(ParserError::UnexpectedToken {
-                    expected: TokenType::Value,
-                    actual: token,
+                return Err(ParserError {
+                    error_kind: ParserErrorKind::UnexpectedToken {
+                        expected: TokenType::Value,
+                        actual: token.kind,
+                    },
+                    span: token.span,
                 });
             }
         };
         self.track_stack_depth_decrease()?;
-        Ok(value)
+        Ok(ValueWithSpan {
+            value,
+            span: token.span,
```

Comment:
Agree, in case of composite Value variants, eg. `Enum`, `Tuple` span information of the delimiters such as (, ) or <, >, commas, discriminators is not stored.
This is acceptable, since we still are able to produce error messages, which are precise enough.